aspnetcore/host-and-deploy/proxy-load-balancer.md

```diff
@@ -0,0 +1,204 @@
+| [KnownProxies](/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersoptions.knownproxies) | Addresses of known proxies to accept forwarded headers from. Use `KnownProxies` to specify exact IP address matches.<br><br>The default is an [IList](/dotnet/api/system.collections.generic.ilist-1)\<[IPNetwork](/dotnet/api/microsoft.aspnetcore.httpoverrides.ipnetwork)> containing a single entry for `IPAddress.IPv6Loopback`. |
+| [OriginalForHeaderName](/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersoptions.originalforheadername) | Use the header specified by this property instead of the one specified by [ForwardedHeadersDefaults.XOriginalForHeaderName](/dotnet/api/microsoft.aspnetcore.httpoverrides.forwardedheadersdefaults.xoriginalforheadername).<br><br>The default is `X-Original-For`. |
+| [OriginalHostHeaderName](/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersoptions.originalhostheadername) | Use the header specified by this property instead of the one specified by [ForwardedHeadersDefaults.XOriginalHostHeaderName](/dotnet/api/microsoft.aspnetcore.httpoverrides.forwardedheadersdefaults.xoriginalhostheadername).<br><br>The default is `X-Original-Host`. |
+| [OriginalProtoHeaderName](/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersoptions.originalprotoheadername) | Use the header specified by this property instead of the one specified by [ForwardedHeadersDefaults.XOriginalProtoHeaderName](/dotnet/api/microsoft.aspnetcore.httpoverrides.forwardedheadersdefaults.xoriginalprotoheadername).<br><br>The default is `X-Original-Proto`. |
+| [RequireHeaderSymmetry](/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersoptions.requireheadersymmetry) | Require the number of header values to be in sync between the [ForwardedHeadersOptions.ForwardedHeaders](/dotnet/api/microsoft.aspnetcore.builder.forwardedheadersoptions.forwardedheaders) being processed.<br><br>The default is `false`. |
+
+## Scenarios and use cases
+
+### When it isn't possible to add forwarded headers and all requests are secure
+
+In some cases, it might not be possible to add forwarded headers to the requests proxied to the app. If the proxy is enforcing that all public external requests are HTTPS, the scheme can be manually set in `Startup.Configure` before using any type of authentication middleware:
+
+```csharp
+app.Use((context, next) =>
+{
+    context.Request.Scheme = "https";
+    return next();
+});
+```
+
+This code can be disabled with an environment variable or other configuration setting in a development or staging environment.
+
+### Deal with path base and proxies that change the request path
+
+[UsePathBaseExtensions.UsePathBase](/dotnet/api/microsoft.aspnetcore.builder.usepathbaseextensions.usepathbase) is used by IIS/ASP.NET Core Module to trim the app base path.
```

Comment:
IIS forwards the path as is, but routing will need to know what the app base path is. IIS Integration uses UsePathBase to split the Path into Path and PathBase so routing can function properly. App developers can also use UsePathBase for other proxies that pass the path through intact but with an app base path.

Reply:
> App developers can also use UsePathBase for other proxies that pass the path through intact but with an app base path.

It doesn't seem like the impl details for IIS are all that important here. I'll try to refactor this into something close to what your last sentence says.

Reply:
Why are we showing the `StartsWithSegments` down below if `UsePathBase` does the same thing under-the-covers?

Reply:
Not setting PathBase is important in that last example.